src/tools/pkcs11-tool.c

```diff
@@ -4449,18 +4463,9 @@ static int test_verify(CK_SESSION_HANDLE sess)
 	rv = p11->C_GetSessionInfo(sess, &sessionInfo);
 	if (rv != CKR_OK)
 		p11_fatal("C_OpenSession", rv);
-	if ((sessionInfo.state & CKS_RO_USER_FUNCTIONS) == 0) {
-		if (opt_login) {
-			int r;
-			r = login(sess, CKU_CONTEXT_SPECIFIC);
-			if (r != 0){
-				printf("Verify: failed to login, skipping verify tests\n");
-				return errors;
-			}
-		} else {
-			printf("Verify: not logged in, skipping verify tests\n");
-			return errors;
-		}
+	if (!(sessionInfo.state & CKS_RW_USER_FUNCTIONS)) {
+		printf("Verify: not a R/W session, skipping verify tests\n");
```

Comment:
This is a revert commit because this code doesn't have any effect. Checking for a RW session doesn't check if we are currently logged in (if a operation that requires a login will succeed). This test is equivalent to opt_login==true && opt_login_type == CKU_USER. So it only checks that a login was made in the beginning, and so we should run this test. If the current login expires it will still fail with a error. This checks are only here to skip tests that should not be attempted if the user didn't used the option --login.